### PR TITLE
Add Content-Type check for metadata response

### PR DESCRIPTION
**Pull request by yunwwang**
This PR add checking Content-Type value for the metadata response. The Content-Type shall be 'application/fhir+json' according to http://hl7.org/fhir/http.html#mime-type

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR:
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/onc_program/bulk_data_export_sequence.rb [hunk 1]
```diff
@@ -39,6 +39,7 @@ def check_capability_statement
         headers = { accept: 'application/fhir+json' }
         reply = LoggedRestClient.get(url, headers)
```
Comment (arscan) [inline lib/modules/onc_program/bulk_data_export_sequence.rb:40]: We should use the fhir_client here and the built in capabilities functionality at this point, I think.

File: lib/modules/us_core_guidance/test/us_core_r4_capability_statement_sequence_test.rb [hunk 2]
```diff
@@ -25,6 +25,20 @@
 
       assert test_result.fail?, "Test result should be fail but is #{test_result.result}"
     end
+
+    it 'fail if Content-Type is not application/fhir+json' do
```
Comment (arscan) [inline lib/modules/us_core_guidance/test/us_core_r4_capability_statement_sequence_test.rb:29]: A minor thing, but I think the convention is to be "fails if...", so the line reads like "it fails if".  I don't feel strongly about it though.  But If you change it, change all the others too.

**Comment by arscan** [comment]
Could you update the description of this specific test to state the capabilities interaction conforms to the requirements in the base FHIR specification, with a link maybe?

https://github.com/onc-healthit/inferno-program/blob/2e5f51ccf322f8487c910352bbc073127c15062a/lib/modules/onc_program/bulk_data_export_sequence.rb#L159

If someone fails and objects, I want the description of the test be where they can look to find out why they failed.  right now the description just says that it must state support for the bulk export operation, which is not quite the same thing.

**Comment by arscan** [comment]
Is this test consistent with the capabilities test for single patient / smart discovery?  Please make sure that this is completely consistent everywhere we are performing the capabilities interaction on the server & update those other spots if needed.  I thought we needed to update those other spots too, but I could be wrong.

File: lib/modules/onc_program/bulk_data_export_sequence.rb (hunk 1 above)
Comment (yunwwang) [reply lib/modules/onc_program/bulk_data_export_sequence.rb:40]: fixed

**Comment by yunwwang** [comment]
Will continue working on this
